File: 백준/Silver/10773. 제로/제로.java

import java.io.*;
import java.util.*;


public class Main {
    int n,answer;

    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    StringTokenizer st;

    Stack<Integer> stk = new Stack<>();

    void solution() throws Exception{
        n = Integer.parseInt(br.readLine());
        for(int i=0;i<n;i++){
            int num = Integer.parseInt(br.readLine());
            if(num==0){
                if(!stk.isEmpty()){
                    stk.pop();
                }
            }
            else{
                stk.push(num);
            }
        }
        for(int x:stk){
            answer+=x;
        }
        System.out.println(answer);
    }



    public static void main(String[] args) throws Exception {
        new Main().solution();
    }
}

File: 백준/Silver/10773. 제로/제로.cc
#include<iostream>
#include<stack>
#include<vector>
#pragma warning (disable:4996)

using namespace std;

int n, res;
int num[100000];
stack<int> stk;

int main()
{

	scanf("%d", &n);

	for (int i = 0; i < n; i++) {
		scanf("%d", &num[i]);
		if (num[i] != 0) {
			stk.push(num[i]);
		}
		else {
			stk.pop();
		}
	}
	int size = stk.size();
	for (int i = 0; i < size; i++) {
		res += stk.top();
		stk.pop();
	}
	printf("%d\n", res);


	return 0;
}
